### PR TITLE
Add hardware dialog

### LINKED ISSUE #8467: New view info about connected HARDWARE- no friendly
Odroid N2

5.12

Hi. From some version was implement new view of Hardware. New view is not friendly for user. Searching link to device it is very hard. Previous view was better for user. Maybe now view should be  grouped in some table or something to better view for user.... Please think about it.
<!--

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Add a dedicated dialog to show hardware, this dialog replaces the current generic markdown dialog that was used here to enhance the user experience when looking for attached hardware.

Old:
image

New:
image

If advanced mode is not enabled, we only show tty, gpio and input subsystem devices
image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #8467
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: hassio/src/dialogs/hardware/dialog-hassio-hardware.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+            )}
+          >
+          </search-input>
+        </div>
+
+        ${devices
+          .sort((a, b) => compare(a.name, b.name))
+          .map(
+            (device) =>
+              html`<ha-expansion-panel .header=${device.name} outlined>
```
Comment (bramkragten) [inline hassio/src/dialogs/hardware/dialog-hassio-hardware.ts:92]: Maybe include `by id` in the header?

File: hassio/src/dialogs/hardware/dialog-hassio-hardware.ts [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+import { HomeAssistant } from "../../../../src/types";
+import { HassioHardwareDialogParams } from "./show-dialog-hassio-hardware";
+
+const _filterDevices = memoizeOne(
+  (showAdvanced: boolean, hardware: HassioHardwareInfo, filter: string) =>
+    hardware.devices
+      .filter(
+        (device) =>
+          (showAdvanced ||
+            ["tty", "gpio", "input"].includes(device.subsystem)) &&
```
Comment (bramkragten) [inline hassio/src/dialogs/hardware/dialog-hassio-hardware.ts:22]: Maybe also include `sound` to non advanced?

File: hassio/src/dialogs/hardware/dialog-hassio-hardware.ts [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        .heading=${true}
+      >
+        <div class="header" slot="heading">
+          <h2>
+            ${this._dialogParams.supervisor.localize("dialog.hardware.title")}
+          </h2>
+          <mwc-icon-button dialogAction="close">
+            <ha-svg-icon .path=${mdiClose}></ha-svg-icon>
+          </mwc-icon-button>
+          <search-input
```
Comment (bramkragten) [inline hassio/src/dialogs/hardware/dialog-hassio-hardware.ts:76]: We should give this some margin to the sides

File: hassio/src/dialogs/hardware/dialog-hassio-hardware.ts [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+            @value-changed=${this._handleSearchChange}
+            .label=${this._dialogParams.supervisor.localize(
+              "dialog.hardware.search"
+            )}
+          >
+          </search-input>
+        </div>
+
+        ${devices
+          .sort((a, b) => compare(a.name, b.name))
```
Comment (bramkragten) [inline hassio/src/dialogs/hardware/dialog-hassio-hardware.ts:89]: Do the sorting in `_filterDevices`

File: hassio/src/system/hassio-host-info.ts [hunk 5]
```diff
@@ -229,20 +229,20 @@ class HassioHostInfo extends LitElement {
...
+      hardware = await fetchHassioHardwareInfo(this.hass);
     } catch (err) {
-      showAlertDialog(this, {
+      await showAlertDialog(this, {
         title: this.supervisor.localize(
           "system.host.failed_to_get_hardware_list"
         ),
         text: extractApiErrorMessage(err),
       });
+      await nextRender();
```
Comment (bramkragten) [inline hassio/src/system/hassio-host-info.ts:242]: No need for this I think?

File: hassio/src/dialogs/hardware/dialog-hassio-hardware.ts (hunk 2 above)
Comment (ludeeus) [reply hassio/src/dialogs/hardware/dialog-hassio-hardware.ts:22]: When do you use that?
For aduio we have a dedicated API that is mapped to add-ons that support it, and core have no device settings for sound

File: hassio/src/dialogs/hardware/dialog-hassio-hardware.ts (hunk 1 above)
Comment (ludeeus) [reply hassio/src/dialogs/hardware/dialog-hassio-hardware.ts]: That can be quite long

Comment (bramkragten) [reply hassio/src/dialogs/hardware/dialog-hassio-hardware.ts]: Just use ellipsis?

Comment (ludeeus) [reply hassio/src/dialogs/hardware/dialog-hassio-hardware.ts]: If not the entire string is showing, why show anything?
It's very few that even has it

Comment (bramkragten) [reply hassio/src/dialogs/hardware/dialog-hassio-hardware.ts]: I guess hardware is mostly used to get zigbee/zwave/try sticks addresses, and they all have an ID?

Maybe we should use something else, but a user now still needs to open every tty entry to find the right one, would be nice if we showed a better name then what number it got...

File: src/components/ha-expansion-panel.ts [hunk 6]
```diff
@@ -14,12 +14,17 @@ class HaExpansionPanel extends LitElement {
...
+
   @query(".container") private _container!: HTMLDivElement;
 
   protected render(): TemplateResult {
     return html`
       <div class="summary" @click=${this._toggleContainer}>
-        <slot name="header">${this.header}</slot>
+        <slot class="header" name="header">
+          ${this.header}
+          <slot class="secondary" name="secondary">${this.secondary} </slot>
```
Comment (bramkragten) [inline src/components/ha-expansion-panel.ts:26]:
```suggestion
          <slot class="secondary" name="secondary">${this.secondary}</slot>
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**